bodyshopbidsdotcom/git-scripts
# w252659874 Procfile for Worker Instances

Closes https://www.wrike.com/open.htm?id=252659874

# Features

## Update Run Requirements
Multicontainer utilizes ECS, which enables us to easily get environment details from the ECS Task Definitions. This means we no longer need to execute commands via SSM, increasing reliability by eliminating text parsing for environment details and polling for the result of remote command execution.

## Updated Environment Types
Each process in a `Procfile` will be given its own elastic beanstalk environment. Each environment will enable scaling a single container-type independent of the others.

## Multi-deploy
This will deploy the image to each environment type, specifying the command that should execute via a configuration file (Dockerrun.aws.json).

## Expectations
* All environments must be created ahead of time.
* All environments must have their environment variables updated independently.
* This requires 

# Testing
* Check out this branch from git-scripts.
* Checkout the [snapsheet-url-shortener](https://github.com/bodyshopbidsdotcom/snapsheet-url-shortener) and switch to the branch `dockerize`.
* Push to the stage environment `git_push_eb stage`
* Try to run a command that connects to the dataase: `git_run_eb stage rails c`
* [Check in ECS](https://us-west-2.console.aws.amazon.com/ecs/home?region=us-west-2#/clusters) to see that worker environments are running.

![screen shot 2018-07-26 at 2 50 03 pm](https://user-images.githubusercontent.com/177431/43285125-9baca4f0-90e3-11e8-8414-064490f38a3f.png)
